decoder/ih264d_sei.c

```diff
@@ -851,6 +851,18 @@ WORD32 ih264d_parse_fgc(dec_bit_stream_t *ps_bitstrm, dec_struct_t *ps_dec,
                     return ERROR_INV_SEI_FGC_PARAMS;
                 }
 
+                /* Although the fag end of both the NALU and the bitstream buffer */
+                /* is being parsed, not all FGC SEI symbols would have been */
+                /* decoded semantically. The code below detects this condition */
+                if((ps_bitstrm->u4_ofst +
+                    ps_sei->s_sei_fgc_params.au1_num_intensity_intervals_minus1[c] *
+                        (8 + 8 +
+                         (ps_sei->s_sei_fgc_params.au1_num_model_values_minus1[c] + 1) * 16)) >=
```

Comment:
I had assumed sev to be 16 bits because the symbol being read is model value which takes requires 16 bits for decoding exponential goulomb code for 255. But, I will modify as per your directions.